### PR TITLE
Add Resgen Module

### PR DESCRIPTION
**Pull request by darkevilmac**
Remade due to issues on my part. Sorry.

### CODE REVIEW

File: src/main/java/com/elytradev/concrete/resgen/ItemModelResourceProvider.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+	/**
+	 * Can this provider provide a resource for the given name?
+	 *
+	 * @param name the resource name.
+	 * @return true if the required resource can be provided, false otherwise.
+	 */
+	@Override
+	public boolean canProvide(String name) {
+		return name.startsWith(("assets/" + modID + "/models/item/")) && name.endsWith(".json");
```
Comment (exaskye) [inline src/main/java/com/elytradev/concrete/resgen/ItemModelResourceProvider.java]: Extraneous parentheses

**Comment by ghost** [comment]
Please see darkevilmac/Concrete#2

File: src/main/java/com/elytradev/concrete/resgen/ConcreteResourcePack.java [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+import com.elytradev.concrete.common.ConcreteLog;
+import com.elytradev.concrete.reflect.accessor.Accessor;
+import com.elytradev.concrete.reflect.accessor.Accessors;
+import com.elytradev.concrete.reflect.invoker.Invoker;
+import com.elytradev.concrete.reflect.invoker.Invokers;
+import com.google.common.collect.HashBiMap;
+import com.google.common.collect.Lists;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.client.resources.*;
```
Comment (exaskye) [inline src/main/java/com/elytradev/concrete/resgen/ConcreteResourcePack.java]: No

File: src/main/java/com/elytradev/concrete/resgen/ConcreteResourcePack.java [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+	/**
+	 * Check if the place provided matches the location validation.
+	 *
+	 * @param place      The place to check.
+	 * @param validation The validation to use.
+	 * @return true if matches, false otherwise.
+	 */
+	private boolean isLocation(String place, String validation) {
+		return place.startsWith(("assets/" + modID + validation)) && place.endsWith(".json");
```
Comment (exaskye) [inline src/main/java/com/elytradev/concrete/resgen/ConcreteResourcePack.java]: Extraneous parentheses

File: src/main/java/com/elytradev/concrete/resgen/ResourceProvider.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	protected final String modID;
+	protected ConcreteResourcePack resourcePack;
+
+	public ResourceProvider(ConcreteResourcePack resourcePack) {
+		this.modID = resourcePack.modID;
+		this.resourcePack = resourcePack;
+	}
+
+	/**
+	 * Can this provider provide a resource for the given name?
```
Comment (exaskye) [inline src/main/java/com/elytradev/concrete/resgen/ResourceProvider.java:47]: how many resources would a provider provide if a provider could provide resources

**Review by exaskye — APPROVED** [review]

**Comment by ghost** [comment]
I failed my job... Please see darkevilmac/Concrete#3

**Pull request merged**